src/pages/experiments/[experimentId]/plots-and-tables/dot-plot/index.jsx

```diff
@@ -71,75 +89,249 @@ const DotPlotPage = (props) => {
+    if (filterRootNode === 'All') return true;
+
+    const filterClusterCellIds = Array.from(cellSetProperties[filterKey].cellIds);
+
+    const baseClusterParent = cellSetHierarcy.find((cellSet) => cellSet.key === baseCluster);
+    const baseClusterKeys = baseClusterParent.children.map((child) => child.key);
+    const baseClusterCellIds = baseClusterKeys.map((key) => cellSetProperties[key].cellIds);
+
+    // Check if there is at least 2 baseClusters with elements common to filterCluster
+    let numIntersections = 0;
+    return baseClusterCellIds.some((baseCellIds) => {
+      if (filterClusterCellIds.some((filterCellId) => baseCellIds.has(filterCellId))) {
+        numIntersections += 1;
+      }
+
+      return numIntersections === 2;
+    });
+  };
+
+  const hasGroups = useMemo(
+    () => hasGroupsToCompare(config?.selectedCellSet, config?.selectedPoints),
+    [config?.selectedCellSet, config?.selectedPoints, cellSetsLoading],
+  );
+
   useEffect(() => {
-    if (config && config.selectedGenes.length === 0 && !genesFetching && !highestDispersionGenes) {
-      const PROPERTIES = ['dispersions'];
-      const tableState = {
-        pagination: {
-          current: 1, pageSize: config.nMarkerGenes, showSizeChanger: true, total: 0,
-        },
-        geneNamesFilter: null,
-        sorter: { field: PROPERTIES[0], columnKey: PROPERTIES[0], order: 'descend' },
-      };
+    if (cellSetsLoading) return;
 
-      dispatch(loadPaginatedGeneProperties(experimentId, PROPERTIES, plotUuid, tableState));
+    // Marker genes calculation needs that the cellIds in groupBy (refer to fn definition)
+    // be represented by more than one groups in filterBy to enable comparison
+    // if (config?.useMarkerGenes && !hasGroupsToCompare(config.selectedCellSet, config.selectedPoints)) {
+    if (config?.useMarkerGenes && !hasGroups) {
+      setMoreThanTwoGroups(false);
+      return;
     }
-  }, [highestDispersionGenes, config, genesFetching]);
+
+    setMoreThanTwoGroups(true);
+
+    const currentComparedConfig = getComparedConfig(config);
+    if (config && !_.isEqual(previousComparedConfig.current, currentComparedConfig)) {
+      previousComparedConfig.current = currentComparedConfig;
+      dispatch(fetchPlotDataWork(experimentId, plotUuid, plotType));
+    }
+  }, [config, cellSetProperties]);
+
+  const loadInitialCustomGenes = () => {
+    const PROPERTIES = ['dispersions'];
+    const tableState = {
+      pagination: {
+        current: 1, pageSize: config.nMarkerGenes, showSizeChanger: true, total: 0,
+      },
+      geneNamesFilter: null,
+      sorter: { field: PROPERTIES[0], columnKey: PROPERTIES[0], order: 'descend' },
+    };
+
+    dispatch(loadPaginatedGeneProperties(experimentId, PROPERTIES, plotUuid, tableState));
+  };
 
   useEffect(() => {
+    if (config?.selectedGenes.length === 0 && !highestDispersionGenes?.length > 0 && !genesFetching) {
```

Comment:
What is this expression - `!highestDispersionGenes?.length > 0`? maybe you mean `highestDispersionGenes.length`

Reply:
We want this check to pass if `highestDispersionGene` is not loaded.

`!highestDispersionGenes?.length > 0` is a combination of 2 checks : 
-if `highestDispersionGenes` is `undefined`, then it evaluates to `!undefined` which is true
-if `heighestDispersionGenes` is defined and the data is still loading (`length` is 0), then it evaluates to `!([].length > 0)` will evaluate to `true`

Reply:
yeah but !highestDispersionGenes?.length will give the same output in those cases

Reply:
Yup, which is what we want. I just checked `loadPaginatedGeneProperties`. If `highestDispersionGenes` is undefined (first call), it is the call to initialize the state. On error, it populates `data` to an empty array. So we also want to be able to re-dispatch this call if there is a previous error (`data` is empty).